src-web/components/kappnav/DetailView.jsx

```diff
@@ -17,60 +17,87 @@
  *****************************************************************/
 
 import 'carbon-components/scss/globals/scss/styles.scss'
-import React, { Component } from 'react'
+import React, {Component} from 'react';
+import { connect } from 'react-redux';
 import PropTypes from 'prop-types'
 import { Loading } from 'carbon-components-react'
 import StructuredListModule from './common/StructuredListModule'
 import {updateSecondaryHeader, getOverflowMenu, getStatus} from '../../actions/common'
+import SecondaryHeader from './common/SecondaryHeader.jsx'
+import getResourceData from '../../definitions/index';
+import { CONTEXT_PATH, RESOURCE_TYPES } from '../../actions/constants';
+import msgs from '../../../nls/kappnav.properties'
 
 
-class DetailView extends React.Component {
+class DetailView extends Component {
   constructor (props){
     super(props)
 
     this.state = {
       data: {},
-      loading: true
+      loading: true,
+      name : decodeURIComponent(location.pathname.split('/').filter(function (e) { return e })[4]),
+      staticResourceData: (location.pathname.split('/').filter(function(e){return e})[3] === 'was-traditional-app') ? getResourceData(RESOURCE_TYPES.WASTRADITIONALAPP) : (location.pathname.split('/').filter(function(e){return e})[3] === 'liberty-app') ? getResourceData(RESOURCE_TYPES.LIBERTYAPP) : getResourceData(RESOURCE_TYPES.WASNDCELL) 
```

Comment:
These lines need to be refactored/rewritten for maintainability.

I do not think we should have references to `was` or `liberty` specific resources in kAppNav